doc-generator/src/main/scala/Main.scala

```diff
@@ -0,0 +1,135 @@
+import scala.sys.process._
+import better.files.Dsl._
+import com.codacy.plugins.api.results.Pattern
+import com.codacy.plugins.api.results.Result.Level
+import com.codacy.plugins.api.results.Tool
+import play.api.libs.json.Json
+import com.codacy.plugins.api.results.Pattern.Subcategory
+import com.codacy.plugins.api.results.Pattern.Description
+import com.codacy.plugins.api.results.Pattern.DescriptionText
+
+object Main extends App {
+
+  def toInputStream(s: String) =
+    new java.io.ByteArrayInputStream(s.getBytes(java.nio.charset.StandardCharsets.UTF_8.name))
+
+  val targetDir = pwd / "doc-generator" / "target"
+  val clangExtraDir = targetDir / "clang-tools-extra"
+
+  if (!(clangExtraDir).exists && clangExtraDir.nonEmpty) {
+    val pb = Process(
+      command = Seq("git", "clone", "--depth", "1", "https://github.com/llvm-mirror/clang-tools-extra.git"),
+      cwd = Some(targetDir.toJava)
+    )
+    require(pb.! == 0, "git clone failed")
+  } else {
+    val pb = Process(command = Seq("git", "pull"), cwd = Some(clangExtraDir.toJava))
+    require(pb.! == 0, "git pull failed")
+  }
+
+  def isNotTitle(s: String): Boolean = !s.startsWith("#")
+
+  def fixRepoLinks(s: String): String = {
+    val linkRegex = """\[(.+)\]\((.+\.html)\)""".r("name", "link")
+    linkRegex.replaceAllIn(s, matching => {
+      val name = matching.group("name")
+      val link = matching.group("link")
+      val linkToClangTidy =
+        if (link.matches("https?:\\/\\/.+")) link else s"https://clang.llvm.org/extra/clang-tidy/checks/${link.stripPrefix("/")}"
+      val linkEscaped = linkToClangTidy.replaceAllLiterally(" ", "").trim
+      val nameEscaped = name.trim
+      s"[$nameEscaped]($linkEscaped)"
+    })
+  }
+
+  val patternsWithDocs: Seq[(String, String)] = {
+    val iterator = for {
+      file <- (clangExtraDir / "docs" / "clang-tidy" / "checks").children
+      patternId = file.nameWithoutExtension(includeAll = false)
+      if file.extension.exists(_ == ".rst") && file.nameWithoutExtension != "list"
+      markdownFile = Seq("pandoc", "-t", "commonmark", file.pathAsString).!!
+      content = markdownFile.linesIterator.dropWhile(isNotTitle).mkString(System.lineSeparator())
+      withRepoLinks = fixRepoLinks(content)
+    } yield (patternId, withRepoLinks)
+    iterator.toSeq
```

Comment:
Isn't this Scala 2.13? Shouldn't this be `to(Seq)`?

Reply:
I think `.toSeq` is faster but not sure. Let me check.

Reply:
If we are going for performance then we should use a view and then do `to(Seq)` at the end. So that we don't iterate twice through the list

Reply:
Scala Documentation says that it [might be faster](https://www.scala-lang.org/api/2.13.1/scala/collection/Iterator.html#toSeq:Seq[A])

Reply:
But if you use a view it will only iterate once

Reply:
I was not not going for performance. I just use `.toSeq` since it's a `Iterator` native method.
There is no `.view` in `Iterator` since it is already lazy and doesn't need to be converted to a `View`.

Reply:
Makes sense. Performance is not really a concern in the generator anyways